ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/Row.java

```diff
@@ -0,0 +1,126 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client;
+
+import java.util.List;
+
+/**
+ * A single record, returned as part of a query result.
+ */
+public interface Row {
+
+  List<String> columnNames();
+
+  List<String> columnTypes();
+
+  List<Object> values();
+
+  /**
+   * Get the value for a particular column of the Row as an Object.
+   *
+   * @param columnIndex index of column (1-indexed).
+   * @return column value.
+   */
+  Object getObject(int columnIndex);
+
+  /**
+   * Get the value for a particular column of the Row as an Object.
+   *
+   * @param columnName name of column.
+   * @return column value.
+   */
+  Object getObject(String columnName);
+
+  /**
+   * Get the value for a particular column of the Row as a string.
+   *
+   * @param columnIndex index of column (1-indexed).
+   * @return column value.
+   */
+  String getString(int columnIndex);
+
+  /**
+   * Get the value for a particular column of the Row as a string.
+   *
+   * @param columnName name of column.
+   * @return column value.
+   */
+  String getString(String columnName);
+
+  /**
+   * Get the value for a particular column of the Row as an integer.
+   *
+   * @param columnIndex index of column (1-indexed).
+   * @return column value.
+   */
+  Integer getInt(int columnIndex);
+
+  /**
+   * Get the value for a particular column of the Row as an integer.
+   *
+   * @param columnName name of column.
+   * @return column value.
+   */
+  Integer getInt(String columnName);
+
+  /**
+   * Get the value for a particular column of the Row as a long.
+   *
+   * @param columnIndex index of column (1-indexed).
+   * @return column value.
+   */
+  Long getLong(int columnIndex);
+
+  /**
+   * Get the value for a particular column of the Row as a long.
+   *
+   * @param columnName name of column.
+   * @return column value.
+   */
+  Long getLong(String columnName);
+
+  /**
+   * Get the value for a particular column of the Row as a double.
+   *
+   * @param columnIndex index of column (1-indexed).
+   * @return column value.
+   */
+  Double getDouble(int columnIndex);
+
+  /**
+   * Get the value for a particular column of the Row as a double.
+   *
+   * @param columnName name of column.
+   * @return column value.
+   */
+  Double getDouble(String columnName);
+
+  /**
+   * Get the value for a particular column of the Row as a boolean.
+   *
+   * @param columnIndex index of column (1-indexed).
+   * @return column value.
+   */
+  Boolean getBoolean(int columnIndex);
+
+  /**
+   * Get the value for a particular column of the Row as a boolean.
+   *
+   * @param columnName name of column.
+   * @return column value.
+   */
+  Boolean getBoolean(String columnName);
```

Comment:
I'd originally wanted to add similar methods for getting decimals but that requires either:
- parsing the schema to find the precision and scale
- asking the caller to specify precision and scale in the getter
and neither seems great. Is it worth updating the endpoint to return the schema in a more structured form? Feels maybe like overkill.

I also considered adding getter methods for arrays (lists) and maps but I'm not sure how valuable those methods would be without parsing the schema for specific subtypes.

Reply:
Personally I think it's ok to expose the Vert.x types, or if you prefer to wrap them in our own type and delegate internally. Either way really. But I think it would be a shame to lose the functionality that those classes have which I think would be useful to users. E.g. type safe getters, easy conversion to JSON string, conversion to buffers etc.

Reply:
As discussed in KLIP-26, I plan to introduce types that wrap the Vert.x types. Stand by for a follow-up PR with this change.